src/App.js

```diff
@@ -1,21 +1,83 @@
 /* global __DEV__ */
-import React from 'react';
-import { View, StatusBar } from 'react-native';
+import React, { Component } from 'react';
+import { View, StatusBar, Linking } from 'react-native';
 import { Provider, connect } from 'react-redux';
 import { identity } from 'lodash';
 
 import codePush from 'react-native-code-push';
+import OneSignal from 'react-native-onesignal';
 import PropTypes from 'prop-types';
+import { NavigationActions } from 'react-navigation';
 import store from './store/config';
 import Root from './Router';
+import * as types from './store/types';
+import { markNotifications } from './store/feed/actions';
 
 // eslint-disable-next-line
 console.disableYellowBox = true;
 
+class App extends Component {
+  componentWillMount() {
+    OneSignal.inFocusDisplaying(2);
+    OneSignal.addEventListener('ids', this.onIds);
+    Linking.addEventListener('url', this.onUrl);
+  }
+
+  componentWillUnmount() {
+    OneSignal.removeEventListener('ids', this.onIds);
+    Linking.removeEventListener('url', this.onUrl);
+  }
+
+  onIds(device) {
+    store.dispatch({ type: types.ONESIGNAL_ID_RECEIVED, payload: device.userId });
+  }
+
+  onUrl({ url }) {
+    const [path, query] = url.replace('https://kitsu.io/', '').split('?');
+    const paths = path.split('/');
+    const params = query.split('&').map(param => param.split('=')).reduce((acc, curr) => ({
```

Comment:
I'm pretty sure you can use the URL api: https://developer.mozilla.org/en/docs/Web/API/URL, which would definitely make this code a bit more obvious.